server/src/main/java/org/elasticsearch/search/aggregations/metrics/TopHitsAggregatorFactory.java

```diff
@@ -67,7 +66,7 @@ class TopHitsAggregatorFactory extends AggregatorFactory {
     ) throws IOException {
         super(name, context, parent, subFactories, metadata);
         if (context.isInSortOrderExecutionRequired()) {
-            throw new AggregationExecutionException("Top hits aggregations cannot be used together with time series aggregations");
+            throw new IllegalArgumentException("Top hits aggregations cannot be used together with time series aggregations");
```

Comment:
The time series incompatibility should probably also be refactored into `AggregationErrors`